tests/core/engine_adapter/integration/test_integration.py

```diff
@@ -1314,8 +1314,6 @@ def test_sushi(ctx: TestContext, tmp_path_factory: pytest.TempPathFactory):
 
     current_gateway_config = config.gateways[ctx.gateway]
     current_gateway_config.state_schema = sushi_state_schema
-    if (sc := current_gateway_config.state_connection) and sc.type_ == "duckdb":
```

Comment:
This is the main test for this PR.  These guards were introduced originally to prevent the integration tests from hanging